### ISSUE TITLE
Run WMCore central services validation for HG2209

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#11225]
**Impact of the new feature**
WMCore central services

**Is your feature request related to a problem? Please describe.**
Monthly task

**Describe the solution you'd like**
Validate central services in cmsweb-testbed (well, it might have to be in one of our VMs due to the current Rucio setup) and provide the final feedback by the June deadline specified by the CMSWEB team.

It also includes the creation of the service release notes and the validation check-list twiki.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by amaltaro** [comment]
@todor-ivanov the CMSWEB team just completed the deployment of WMCore services in testbed, using tag `HG2209b`. Could you please work on this validation (results due to Friday night)?

**Comment by todor-ivanov** [comment]
Hi @amaltaro I'll take care of that but no way for those workflows to complete before Monday.

**Comment by todor-ivanov** [comment]
Just for logging purposes, here is the error I face every time I try to move a workflow from `assignment-approved` to `assigned` with the new release:

```
[04/Sep/2022:09:10:45] reqmgr2-7689bd755d-p52m8 188.185.14.143 "GET /reqmgr2/data/request?name=spiga_ReReco_339_TB_CMS_Marconi_220731_175607_3091 HTTP/1.1" 200 OK [data: 3936 in 1925 out 26215 us ] [auth: OK "/DC=ch/DC=cern/OU=computers/CN=wmagent/vocms0192.cern.ch" "" ] [ref: "" "WMCore.Services.Requests/v002" ]
[04/Sep/2022:09:10:45]  Updating request "tivanov_ACDC_TaskChain_LumiMask_multiRun_HG2209_Val_220904_091028_2526" with these user-provided args: {'RequestStatus': 'assigned', 'RequestPriority': 200000, 'Team': 'testbed-vocms0192', 'SiteWhitelist': ['T1_US_FNAL', 'T2_CH_CERN'], 'SiteBlacklist': '', 'AcquisitionEra': {'HLTD': 'CMSSW_7_2_0', 'RECODreHLT': 'CMSSW_7_2_0'}, 'ProcessingString': {'HLTD': 'HLTD_TaskChain_LumiMask_multiRun_HG2209_Val_Todor_v1', 'RECODreHLT': 'RECODreHLT_TaskChain_LumiMask_multiRun_HG2209_Val_Todor_v1'}, 'ProcessingVersion': 11, 'Dashboard': 'integration', 'MergedLFNBase': '/store/mc', 'TrustSitelists': 'False', 'UnmergedLFNBase': '/store/unmerged', 'MinMergeSize': 2147483648, 'MaxMergeSize': 4294967296, 'MaxMergeEvents': 100000, 'BlockCloseMaxWaitTime': 72000, 'BlockCloseMaxFiles': 500, 'BlockCloseMaxEvents': 250000000, 'BlockCloseMaxSize': 5000000000000, 'SoftTimeout': 129600, 'GracePeriod': 1000, 'TrustPUSitelists': 'False', 'CustodialSites': '', 'CustodialSubType': '', 'NonCustodialSites': '', 'NonCustodialSubType': '', 'Override': {'eos-lfn-prefix': 'root://eoscms.cern.ch//eos/cms/store/logs/prod/recent/TESTBED'}, 'AutoApproveSubscriptionSites': '', 'SubscriptionPriority': 'Low'}
[04/Sep/2022:09:10:46]  Error: Traceback (most recent call last):
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/ReqMgr/Service/Request.py", line 188, in validate
    self._validateRequestBase(param, safe, validate_request_update_args, requestName)
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/ReqMgr/Service/Request.py", line 100, in _validateRequestBase
    workload, r_args = valFunc(args, self.config, self.reqmgr_db_service, param)
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/ReqMgr/Utils/Validation.py", line 60, in validate_request_update_args
    workload.validateArgumentForAssignment(request_args)
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkload.py", line 1945, in validateArgumentForAssignment
    validateArgumentsUpdate(schema, argumentDefinition)
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkloadTools.py", line 292, in validateArgumentsUpdate
    validateUnknownArgs(arguments, argumentDefinition)
  File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkloadTools.py", line 323, in validateUnknownArgs
    raise WMSpecFactoryException(msg)
WMCore.WMSpec.WMSpecErrors.WMSpecFactoryException: <@========== WMException Start ==========@>
Exception Class: WMSpecFactoryException
Message: There are unknown/unsupported arguments in your request spec: ['AutoApproveSubscriptionSites', 'NonCustodialSubType', 'CustodialSubType']
        ClassName : None
        ModuleName : WMCore.WMSpec.WMWorkloadTools
        MethodName : validateUnknownArgs
        ClassInstance : None
        FileName : /data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkloadTools.py
        LineNumber : 323
        ErrorNr : 0

Traceback: 

<@---------- WMException End ----------@>

[04/Sep/2022:09:10:46]  SERVER REST ERROR WMCore.ReqMgr.DataStructs.RequestError.InvalidSpecParameterValue 53d67b1934bc2cd41f720c7b170dc0e9 (Invalid spec parameter value: There are unknown/unsupported arguments in your request spec: ['AutoApproveSubscriptionSites', 'NonCustodialSubType', 'CustodialSubType'])
[04/Sep/2022:09:10:46]    Traceback (most recent call last):
[04/Sep/2022:09:10:46]      File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/REST/Server.py", line 728, in default
[04/Sep/2022:09:10:46]        return self._call(RESTArgs(list(args), kwargs))
[04/Sep/2022:09:10:46]      File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/REST/Server.py", line 805, in _call
[04/Sep/2022:09:10:46]        v(apiobj, request.method, api, param, safe)
[04/Sep/2022:09:10:46]      File "/data/srv/HG2209c/sw/slc7_amd64_gcc630/cms/reqmgr2/2.1.2.pre3/lib/python3.8/site-packages/WMCore/ReqMgr/Service/Request.py", line 223, in validate
[04/Sep/2022:09:10:46]        raise InvalidSpecParameterValue(msg) from None
[04/Sep/2022:09:10:46]    WMCore.ReqMgr.DataStructs.RequestError.InvalidSpecParameterValue: InvalidSpecParameterValue 53d67b1934bc2cd41f720c7b170dc0e9 [HTTP 400, APP 1102, MSG "Invalid spec parameter value: There are unknown/unsupported arguments in your request spec: ['AutoApproveSubscriptionSites', 'NonCustodialSubType', 'CustodialSubType']", INFO None, ERR None]

```

**Comment by amaltaro** [comment]
@todor-ivanov this is a blocker for the CMSWEB upgrade. 
Please create a new GH issue and follow this up with the highest priority such that we can still try to have the deployment on Tuesday. This change went in in the last commits and we might to add code to make old and new workflows compatible (if I understand the issue correctly...).

**Comment by todor-ivanov** [comment]
Hi @amaltaro, working on it already. Here is the new issue: https://github.com/dmwm/WMCore/issues/11269 I will ask CMSWeb team to postpone the Tuesday deployment, because I do not see a way for us to properly fix and test it by then.

**Comment by todor-ivanov** [comment]
Upon resolving the previous issue stumbled on another one. I found the GlobalWorkqueue in testbed stuck since yesterday and the only 4 lines we had in the logs for the whole  day were: 
```
[user@vocms0750 dmwm-logs]$ cat  workqueue-20220905-workqueue-7fcdb89f77-jprh9.log 
Document not found: {'key': '630d158254d05bb0d759d69e05761272', 'error': 'not_found'}
Document not found: {'key': 'b050bd43b4cd1958ef972569be48aae9', 'error': 'not_found'}
Document not found: {'key': '837ad6c6409f5d3e28e4bff1beb6c88a', 'error': 'not_found'}
Document not found: {'key': 'bf3f36a6293254f20cfe5dfb937e6abf', 'error': 'not_found'}
```  

Upon logging into the pod directly and restarting it I still see the following error:
```
[06/Sep/2022:12:09:06] ENGINE Bus STARTED
Document not found: {'key': 'a07eb2fdee9c4b43a3b8b98c1a2dfb40', 'error': 'not_found'}
Document not found: {'key': '3b1433be8c58a207c2065792ef688e14', 'error': 'not_found'}
The cachefile /data/srv/state/workqueue/cache/.wmcore_cache_1000/requests/cmsweb-testbed.cern.ch:8443/-6680380037694496850_GET_request?status=aborted&status=force-complete&detail=False does not exist and the service at https://cmsweb-testbed.cern.ch:8443/reqmgr2/data/request?status=aborted&status=force-complete&detail=False is unavailable - it returned 500 because Internal Server Error with result: b'<!DOCTYPE html PUBLIC\n"-//W3C//DTD XHTML 1.0 Transitional//EN"\n"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n<html>\n<head>\n    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"></meta>\n    <title>500 Internal Server Error</title>\n    <style type="text/css">\n    #powered_by {\n        margin-top: 20px;\n        border-top: 2px solid black;\n        font-style: italic;\n    }\n\n    #traceback {\n        color: red;\n    }\n    </style>\n</head>\n    <body>\n        <h2>500 Internal Server Error</h2>\n        <p>Server error</p>\n        <pre id="traceback"></pre>\n    <div id="powered_by">\n      <span>\n        Powered by <a href="http://www.cherrypy.dev">CherryPy 18.8.0</a>\n      </span>\n    </div>\n    </body>\n</html>\n'

The cachefile /data/srv/state/workqueue/cache/.wmcore_cache_1000/requests/cmsweb-testbed.cern.ch:8443/-6680380037694496850_GET_request?status=staged&detail=True does not exist and the service at https://cmsweb-testbed.cern.ch:8443/reqmgr2/data/request?status=staged&detail=True is unavailable - it returned 500 because Internal Server Error with result: b'<!DOCTYPE html PUBLIC\n"-//W3C//DTD XHTML 1.0 Transitional//EN"\n"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n<html>\n<head>\n    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"></meta>\n    <title>500 Internal Server Error</title>\n    <style type="text/css">\n    #powered_by {\n        margin-top: 20px;\n        border-top: 2px solid black;\n        font-style: italic;\n    }\n\n    #traceback {\n        color: red;\n    }\n    </style>\n</head>\n    <body>\n        <h2>500 Internal Server Error</h2>\n        <p>Server error</p>\n        <pre id="traceback"></pre>\n    <div id="powered_by">\n      <span>\n        Powered by <a href="http://www.cherrypy.dev">CherryPy 18.8.0</a>\n      </span>\n    </div>\n    </body>\n</html>\n'

The cachefile /data/srv/state/workqueue/cache/.wmcore_cache_1000/requests/cmsweb-testbed.cern.ch:8443/-6680380037694496850_GET_request?status=aborted&status=force-complete&detail=False does not exist and the service at https://cmsweb-testbed.cern.ch:8443/reqmgr2/data/request?status=aborted&status=force-complete&detail=False is unavailable - it returned 500 because Internal Server Error with result: b'<!DOCTYPE html PUBLIC\n"-//W3C//DTD XHTML 1.0 Transitional//EN"\n"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n<html>\n<head>\n    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"></meta>\n    <title>500 Internal Server Error</title>\n    <style type="text/css">\n    #powered_by {\n        margin-top: 20px;\n        border-top: 2px solid black;\n        font-style: italic;\n    }\n\n    #traceback {\n        color: red;\n    }\n    </style>\n</head>\n    <body>\n        <h2>500 Internal Server Error</h2>\n        <p>Server error</p>\n        <pre id="traceback"></pre>\n    <div id="powered_by">\n      <span>\n        Powered by <a href="http://www.cherrypy.dev">CherryPy 18.8.0</a>\n      </span>\n    </div>\n    </body>\n</html>\n'

The cachefile /data/srv/state/workqueue/cache/.wmcore_cache_1000/requests/cmsweb-testbed.cern.ch:8443/-6680380037694496850_GET_request?status=staged&detail=True does not exist and the service at https://cmsweb-testbed.cern.ch:8443/reqmgr2/data/request?status=staged&detail=True is unavailable - it returned 500 because Internal Server Error with result: b'<!DOCTYPE html PUBLIC\n"-//W3C//DTD XHTML 1.0 Transitional//EN"\n"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n<html>\n<head>\n    <meta http-equiv="Content-Type" content="text/html; charset=utf-8"></meta>\n    <title>500 Internal Server Error</title>\n    <style type="text/css">\n    #powered_by {\n        margin-top: 20px;\n        border-top: 2px solid black;\n        font-style: italic;\n    }\n\n    #traceback {\n        color: red;\n    }\n    </style>\n</head>\n    <body>\n        <h2>500 Internal Server Error</h2>\n        <p>Server error</p>\n        <pre id="traceback"></pre>\n    <div id="powered_by">\n      <span>\n        Powered by <a href="http://www.cherrypy.dev">CherryPy 18.8.0</a>\n      </span>\n    </div>\n    </body>\n</html>\n'
```

The interfaces mentioned above in the log are actually working and are browse-able. (e.g.   https://cmsweb-testbed.cern.ch:8443/reqmgr2/data/request?status=staged&detail=True)

The reason for me looking into the GW in the first place were some of the ACDCs I have created yesterday which were not progressing. They are sitting in `staged` and are not having workqueue elements created: 

https://cmsweb-testbed.cern.ch/reqmgr2/fetch?rid=tivanov_ACDC_ReReco_RunBlockWhite_HG2209_Val_220904_091356_2976

https://cmsweb-testbed.cern.ch/reqmgr2/fetch?rid=tivanov_ACDC_TaskChain_LumiMask_multiRun_HG2209_Val_220904_091028_2526

**Comment by todor-ivanov** [comment]
Yet another workflow: [tivanov_TC_Drop_PhEDEx_Ext_HG2209_Val_220901_112931_5259](https://cmsweb-testbed.cern.ch/reqmgr2/fetch?rid=tivanov_TC_Drop_PhEDEx_Ext_HG2209_Val_220901_112931_5259) found to sit in `running-open` due to some errors in the agent's WorkQueue: [1]. This time it is not due to CouchDB errors, but rather due to a missing configuration parameter for the subscription. 



[1]
```
2022-09-06 14:49:19,191:139924790806272:ERROR:WMBSHelper:Failed to create subscription. Error: 'ConfigSection' object has no attribute 'autoApproveSites'
Traceback (most recent call last):
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 413, in createSubscriptionAndAddFiles
    self.createSubscription(self.topLevelTask, self.topLevelFileset)
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 266, in createSubscription
    self._createDatasetSubscriptionsInDBSBuffer()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 515, in _createDatasetSubscriptionsInDBSBuffer
    subInfo = self.wmSpec.getSubscriptionInformation()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkload.py", line 1561, in getSubscriptionInformation
    taskSubInfo = self.getSubscriptionInformation(task)
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkload.py", line 1556, in getSubscriptionInformation
    subInfo = initialTask.getSubscriptionInformation()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMTask.py", line 1085, in getSubscriptionInformation
    "AutoApproveSites": outputSection.autoApproveSites,
AttributeError: 'ConfigSection' object has no attribute 'autoApproveSites'
2022-09-06 14:49:19,192:139924790806272:ERROR:WorkQueue:Failed to create subscription for tivanov_TC_Drop_PhEDEx_Ext_HG2209_Val_220901_112931_5259 with block name None
Error: 'ConfigSection' object has no attribute 'autoApproveSites'
Traceback (most recent call last):
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WorkQueue.py", line 352, in getWork
    match['Subscription'] = self._wmbsPreparation(match,
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WorkQueue.py", line 453, in _wmbsPreparation
    sub, match['NumOfFilesAdded'] = wmbsHelper.createSubscriptionAndAddFiles(block=dbsBlock)
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 418, in createSubscriptionAndAddFiles
    raise ex
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 413, in createSubscriptionAndAddFiles
    self.createSubscription(self.topLevelTask, self.topLevelFileset)
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 266, in createSubscription
    self._createDatasetSubscriptionsInDBSBuffer()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WorkQueue/WMBSHelper.py", line 515, in _createDatasetSubscriptionsInDBSBuffer
    subInfo = self.wmSpec.getSubscriptionInformation()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkload.py", line 1561, in getSubscriptionInformation
    taskSubInfo = self.getSubscriptionInformation(task)
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMWorkload.py", line 1556, in getSubscriptionInformation
    subInfo = initialTask.getSubscriptionInformation()
  File "/data/srv/wmagent/v2.0.4.patch1/sw/slc7_amd64_gcc630/cms/wmagentpy3/2.0.4.patch1/lib/python3.8/site-packages/WMCore/WMSpec/WMTask.py", line 1085, in getSubscriptionInformation
    "AutoApproveSites": outputSection.autoApproveSites,
AttributeError: 'ConfigSection' object has no attribute 'autoApproveSites'

```

**Comment by todor-ivanov** [comment]
FYI @amaltaro  
I think the errors I am pointing above are not to be ignored for this release. On top of that we have a set of validation workflows that are unable to progress due to old templates and most probably missing input data. Here: [1] is the set of workflows and the relevant templates. 

Even though I am capable to validate what is already in state `completed` I am not sure if we are in good shape for rushing a production deployment tomorrow.

**Comment by amaltaro** [comment]
@todor-ivanov for the ACDCs failing at global workqueue, as we discussed over Zoom, problem is that those ACDCs were created for Harvest jobs, which is not supported (we can create a ticket to deal with that in the future). I understand those have already been aborted.

For the local agent's workqueue, it looks like we have to backport Jhonatan's patch to the `2.0.4_wmagent` branch. I just did (with one extra PR backported) and here it is:
https://github.com/dmwm/WMCore/pull/11273

it has been applied to vocms0192 as well. If everything goes well, we need to apply this same patch to ALL of the agents before the central services get upgraded tomorrow (all the components need to be restarted).
